tests/python/gaia-ui-tests/gaiatest/tests/graphics/test_music_fullplay_song.py

```diff
@@ -0,0 +1,64 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import By, Wait
+
+from gaiatest.gaia_graphics_test import GaiaImageCompareTestCase
+from gaiatest.apps.music.regions.player_view import PlayerView
+from gaiatest.apps.music.app import Music
+
+
+class TestFullPlayMusic(GaiaImageCompareTestCase):
+    _player_controls_previous_locator = (By.ID, 'player-controls-previous')
+    _player_controls_next_locator = (By.ID, 'player-controls-next')
+    _main_song_tile = (By.CSS_SELECTOR, '.main-tile')
+
+    def setUp(self):
+        GaiaImageCompareTestCase.setUp(self)
+
+        # add video to storage
+        self.push_resource('MUS_0001.mp3')
+
+    def test_music_songs_fullplay(self):
+        """
+        https://moztrap.mozilla.org/manage/case/4031/
+        play a song until the end
+        """
+
+        music_app = Music(self.marionette)
+        music_app.launch()
+        music_app.wait_for_music_tiles_displayed()
+        self.take_screenshot()
+
+        # tap a song in tiles view
+        self.marionette.find_element(*self._main_song_tile).tap()
+        from_tile_player_view = PlayerView(self.marionette)
+
+        play_time = time.strptime('00:13', '%M:%S')
+        Wait(self.marionette, timeout=20).until(
+            lambda m: from_tile_player_view.player_elapsed_time == play_time,
+            message='song did not reach the end')
+
+        music_app.wait_for_music_tiles_displayed()
```

Comment:
Because I wanted to make sure that the song plays fine when launched from at least more than one view, in case the bug is specific to where the song was launched from.